protocol.md

```diff
@@ -0,0 +1,143 @@
+# Chat Protocol based on MQTT v5
+
+## Abstract
+
+This document delineates a chat protocol grounded in the MQTT v5 standard, promoting secure and decentralized messaging through MQTT clients. The protocol specifies the topic structure, message format, and operational guidelines to foster compatibility and interoperability across a diverse range of MQTT clients.
```

Comment:
For decentralization, I think the protocol would need to support more than one broker. Otherwise the single broker would be a centralized point in the protocol.

If we want to support more than one broker we we could do this by specifying special mediation clients that acts as a bridge between different MQTT brokers. We also have to decide if we want to go for federation like XMPP, or something more like https://tox.chat/ where clients don't need to connect to a specific server. Federation will be easier.

Maybe we can start to define how the protocol  should work on a single MQTT broker first but keep in mind that we might want to extend it to work with several brokers in the feature?

We could require that the user name has the format `PUBLIC_PGP_KEY@broker.hostname` and that all messages are signed by the `PUBLIC_PGP_KEY` of the user sending that message so that the identity of the sender can be checked by the clients. As a user friendliness feature messages can also include the alias/name of the sender so that chat clients can display that instead of the `PUBLIC_PGP_KEY`. This would also make it easy to implement end-to-end encryption between two chat clients.